modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java

```diff
@@ -1118,10 +1120,12 @@ protected void patchVendorExtensionNullableValueType(CodegenParameter parameter)
     /**
      * Returns the model related to the given parameter
      */
-    private CodegenModel getModelFromParameter(List<ModelMap> allModels, CodegenParameter parameter) {
-        return parameter.isModel || parameter.getIsEnumOrRef()
-                ? allModels.stream().map(m -> m.getModel()).filter(m -> m.getClassname().equals(parameter.dataType)).findFirst().orElse(null)
-                : null;
+    private CodegenModel getModelFromParameter(HashMap<String, CodegenModel> allModels, CodegenParameter parameter) {
+        if (parameter.isModel || parameter.getIsEnumOrRef()) {
+            return allModels.getOrDefault(parameter.dataType, null);
+        } else {
+            return null;
+        }
```

Comment:
@devhl-labs FYI. Minor refactoring to avoid going through the list of modelMap every time the function is called.